gammapy/datasets/map.py

```diff
@@ -339,6 +337,32 @@ def npred(self):
                 npred_total.stack(npred)
         return npred_total
 
+    def npred_sig(self, model=None):
+        """"Model predicted signal counts. If a model is passed, predicted counts from that component is returned.
+        Else, the total signal counts are returned.
+
+        Parameters
+        -------------
+        model: `~gammapy.modeling.models.Models`, optional
+            The model to computed the npred for. If none, the sum of all components (minus the background model)
+            is returned
+
+        Returns
+        ----------
+        npred_sig: `gammapy.maps.WcsNDMap`
+        """
+        if model is None:
+            if self.background_model is None:
+                return self.npred()
+            return self.npred() - self.background_model.evaluate()
+        else:
+            return MapEvaluator(
```

Comment:
I would just rely on `.evaluators[model]` here and not create a new `MapEvaluator`, also the current implementation does not work, because `MapDataset._edisp_kernel` does not exist.